consumer/parsers.go

```diff
@@ -92,3 +100,55 @@ func ParseDeleteMessage(msg []byte, output interface{}) error {
 	}
 	return nil
 }
+
+// setTransactionIdAsIdempotencyKey extracts transaction_id from the message payload and sets it as the IdempotencyKey
+// in the ReportResourceRequest's Representations Metadata if the output is a ReportResourceRequest
+func setTransactionIdAsIdempotencyKey(payload interface{}, output interface{}) error {
+	// Check if the output is a ReportResourceRequest
+	req, ok := output.(*v1beta2.ReportResourceRequest)
+	if !ok {
+		return nil
+	}
+
+	// Extract transaction_id using helper function
+	transactionIdStr, err := extractTransactionId(payload)
+	if err != nil {
+		return err
+	}
+	if transactionIdStr == "" {
+		return nil
+	}
+
+	// Ensure Representations and Metadata exist
+	if req.Representations == nil {
+		req.Representations = &v1beta2.ResourceRepresentations{}
+	}
+	if req.Representations.Metadata == nil {
+		req.Representations.Metadata = &v1beta2.RepresentationMetadata{}
+	}
+
+	// Set the transaction_id as the IdempotencyKey
+	req.Representations.Metadata.IdempotencyKey = &v1beta2.RepresentationMetadata_TransactionId{
+		TransactionId: transactionIdStr,
+	}
+
+	return nil
+}
+
+// extractTransactionId navigates the nested payload structure to find transaction_id
+func extractTransactionId(payload interface{}) (string, error) {
+	// Navigate through payload.representations.metadata.transaction_id
+	payloadMap, ok := payload.(map[string]interface{})
+	if !ok {
+		return "", nil
+	}
+
+	if representations, ok := payloadMap["representations"].(map[string]interface{}); ok {
+		if metadata, ok := representations["metadata"].(map[string]interface{}); ok {
+			if transactionId, ok := metadata["transaction_id"].(string); ok {
+				return transactionId, nil
+			}
+		}
+	}
```

Comment:
nit: we could reduce nesting here to make things more readable, if you agree
```
if representations, ok ...; !ok {
    return "", nil
}

if metadata, ok ...; !ok {
    return "", nil
}

if transactionId, ok ... {
    return transactionId, nil
}
```

Reply:
haha i had it like that originally and i was worried it was maybe too much, im good with flipping it back though